ada/language/ast.py

```diff
@@ -6919,6 +6951,100 @@ def is_write_reference():
             lambda _: False
         )
 
+    @langkit_property(dynamic_vars=[default_imprecise_fallback()])
+    def potential_actuals_for_dispatch():
+        spec = Var(Entity.referenced_decl.subp_spec_or_null)
+
+        # Handle the case where the dispatch is done on the tag of the
+        # LHS expr of an assign statement. In that case, its expected type
+        # is the return type of the called subprogram.
+        # TODO: The assign statement might not be the direct parent, e.g
```

Comment:
So:

1. You're already doing better than GNAT here (as checked live)
2. If you add a qualified expression around the call, the target type will be statically known.
3. With a type conversion, it's no longer a controlling result call (apparently)

So you can proudly remove the TODO ;)